src/include/lru.h

```diff
@@ -157,14 +157,14 @@ class LRU {
 
   // expire -- expire a single item
   LRUObject *lru_get_next_expire() {
+    adjust();
     // look through tail of bot
     while (bottom.size()) {
       LRUObject *p = bottom.back();
       if (!p->lru_pinned) return p;
 
       // move to pintail
       pintail.push_front(&p->lru_link);
-      adjust();
```

Comment:
I believe this `adjust()` is necessary because the size of bottom is changing so it may be necessary to move objects from `top` to `bottom`.